llvm/test/CodeGen/AArch64/GlobalISel/combine-cast-const.mir

```diff
@@ -0,0 +1,45 @@
+# NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py
+# RUN: llc -run-pass=aarch64-prelegalizer-combiner -verify-machineinstrs -mtriple aarch64-unknown-unknown %s -o - | FileCheck %s
+
+---
+name:            zext_const
+body:             |
+  bb.0:
+    liveins: $w0, $w1
+    ; CHECK-LABEL: name: zext_const
+    ; CHECK: liveins: $w0, $w1
+    ; CHECK-NEXT: {{  $}}
+    ; CHECK-NEXT: [[C:%[0-9]+]]:_(s64) = G_CONSTANT i64 5
+    ; CHECK-NEXT: $x0 = COPY [[C]](s64)
+    %2:_(s32) = G_CONSTANT i32 5
+    %3:_(s64) = G_ZEXT %2
+    $x0 = COPY %3
+...
+---
+name:            sext_const
+body:             |
+  bb.0:
+    liveins: $q0, $q1
+    ; CHECK-LABEL: name: sext_const
+    ; CHECK: liveins: $q0, $q1
+    ; CHECK-NEXT: {{  $}}
+    ; CHECK-NEXT: [[C:%[0-9]+]]:_(s64) = G_CONSTANT i64 -5
+    ; CHECK-NEXT: $x0 = COPY [[C]](s64)
+    %2:_(s32) = G_CONSTANT i32 -5
+    %3:_(s64) = G_SEXT %2
+    $x0 = COPY %3
+...
+---
+name:            trunc_const
+body:             |
+  bb.0:
+    liveins: $q0, $q1
+    ; CHECK-LABEL: name: trunc_const
+    ; CHECK: liveins: $q0, $q1
+    ; CHECK-NEXT: {{  $}}
+    ; CHECK-NEXT: [[C:%[0-9]+]]:_(s32) = G_CONSTANT i32 705032704
+    ; CHECK-NEXT: $w0 = COPY [[C]](s32)
+    %2:_(s64) = G_CONSTANT i64 5000000000
+    %3:_(s32) = G_TRUNC %2
+    $w0 = COPY %3
+
```

Comment:
Check vector cases

Reply:
Firstly, the pattern does not match for vector cases. Secondly, to implement the same combine for ext of build vector we need varargs in MIR.

Reply:
I thought splats were handled already

Reply:
and still, showing the negative case for vectors is still useful for the future patch that handles vectors

Reply:
https://github.com/llvm/llvm-project/blob/ef16f7ac1b3267e11bad464b4116d34396e7967f/llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp#L12890

Reply:
https://github.com/llvm/llvm-project/issues/87459